pennylane/ops/op_math/controlled_ops.py

```diff
@@ -117,68 +132,108 @@ class ControlledQubitUnitary(ControlledOp):
+                warnings.warn(
+                    "base operator already has wires; values specified through wires kwarg will be ignored."
+                )
+                wires = Wires(())
 
-        if hasattr(base, "wires") and len(wires) != 0:
-            warnings.warn(
-                "base operator already has wires; values specified through wires kwarg will be ignored."
+            all_wires = control_wires + wires
+            return cls._primitive.bind(
+                base, control_wires=all_wires, control_values=control_values, work_wires=work_wires
             )
-            wires = Wires(())
+        else:
 
-        all_wires = control_wires + wires
-        return cls._primitive.bind(
-            base, control_wires=all_wires, control_values=control_values, work_wires=work_wires
-        )
+            work_wires = Wires(() if work_wires is None else work_wires)
+
+            if hasattr(base, "wires") and len(wires) != 0:
+                warnings.warn(
+                    "base operator already has wires; values specified through wires kwarg will be ignored."
+                )
+                # wires = Wires(())
+
+            return cls._primitive.bind(
+                base, wires=wires, control_values=control_values, work_wires=work_wires
+            )
 
     # pylint: disable=too-many-arguments,too-many-positional-arguments
     def __init__(
         self,
         base,
-        control_wires: WiresLike,
-        wires: WiresLike = (),
+        wires: WiresLike = None,
+        control_wires: WiresLike = "unset",
         control_values=None,
         unitary_check=False,
         work_wires: WiresLike = (),
     ):
-        wires = Wires(() if wires is None else wires)
-        work_wires = Wires(() if work_wires is None else work_wires)
-        control_wires = Wires(control_wires)
-
-        if hasattr(base, "wires") and len(wires) != 0:
-            warnings.warn(
-                "base operator already has wires; values specified through wires kwarg will be ignored."
-            )
-            wires = Wires(())
-
-        if isinstance(base, Iterable):
-            if len(wires) == 0:
-                if len(control_wires) > 1:
-                    num_base_wires = int(qml.math.log2(qml.math.shape(base)[-1]))
-                    wires = control_wires[-num_base_wires:]
-                    control_wires = control_wires[:-num_base_wires]
-                else:
-                    raise TypeError(
-                        "Must specify a set of wires. None is not a valid `wires` label."
-                    )
-            # We use type.__call__ instead of calling the class directly so that we don't bind the
-            # operator primitive when new program capture is enabled
-            base = type.__call__(qml.QubitUnitary, base, wires=wires, unitary_check=unitary_check)
+        _deprecate_control_wires(control_wires)
+        if control_wires != "unset":
```

Comment:
Non-blocking comment for future awareness.  It tends to be easier to read to control on the positive. One less thing to hold in your mind. 

So comparing:
```
if val != "unset":
   # thing one
else:
   # thing two
```
and
```
if val == "unset":
    # thing two
else:
    #thing one 
```
The execption would be if `# thing one` was substantially simpler and shorter than `# thing two`.

Reply:
fixed at https://github.com/PennyLaneAI/pennylane/pull/6839/commits/f8095b140dae400813176fe7b83b47f680fae9c7